src/data_source/storage/sql.rs

```diff
@@ -401,31 +490,42 @@ pub struct Pruner {
 
 impl SqlStorage {
     /// Connect to a remote database.
-    pub async fn connect(mut config: Config) -> Result<Self, Error> {
+    pub async fn connect<DB: Database>(mut config: Config<DB>) -> Result<Self, Error> {
+        let pool = config.pool_opt;
+
+        #[cfg(not(feature = "embedded-db"))]
         let schema = config.schema.clone();
-        let pool = config
-            .pool_opt
-            .after_connect(move |conn, _| {
-                let schema = schema.clone();
-                async move {
-                    query(&format!("SET search_path TO {schema}"))
-                        .execute(conn)
-                        .await?;
-                    Ok(())
-                }
-                .boxed()
-            })
-            .connect_with(config.db_opt)
-            .await?;
+        #[cfg(not(feature = "embedded-db"))]
+        let pool = pool.after_connect(move |conn, _| {
+            let schema = config.schema.clone();
+            async move {
+                query(&format!("SET search_path TO {schema}"))
+                    .execute(conn)
+                    .await?;
+                Ok(())
+            }
+            .boxed()
+        });
+
+        #[cfg(feature = "embedded-db")]
+        if config.reset {
+            std::fs::remove_file(config.db_path)?;
+        }
+
+        let pool = pool.connect(config.db_opt.to_url_lossy().as_ref()).await?;
```

Comment:
Why did we change this back to `to_url_lossy`? Some of the options we care about, such as log levels, don't get encoded in the URL, but do get passed through if we use `connect_with(config.db_opt)`